ecosystem/platform/client/src/ui/SocialLoginButton.tsx

```diff
@@ -0,0 +1,127 @@
+import * as React from "react";
+import {assertNever, randomUUID} from "utils";
+
+const CLIENT_IDS = Object.freeze({
+  github: process.env.REACT_APP_GITHUB_CLIENT_ID,
+});
+
+type Service = keyof typeof CLIENT_IDS;
+
+type Props = {
+  service: Service;
+  onSuccess: () => void;
+  [prop: string]: any;
+};
+
+function oauthUrl(service: Service, csrfToken: string): string {
+  const clientId = CLIENT_IDS[service];
+  switch (service) {
+    case "github":
+      return `https://github.com/login/oauth/authorize?scope=user:email&client_id=${clientId}&state=${csrfToken}`;
+    default:
+      return assertNever(service);
+  }
+}
+
```

Comment:
this is the code I've been using to drive the oauth flow:
```
cookieStore.get("CSRF-TOKEN").then((token)=>{
    var myForm = document.createElement("FORM");
    myForm.setAttribute("id","TestForm");
    document.body.appendChild(myForm);
    myForm.method = "POST";
    myForm.action = "/users/auth/github?authenticity_token=" + token.value;
    myForm.submit();
 });
 ```

Reply:
Yeah, much of this will need to be updated once the server-side code is in.